docs/recipes/isolating-mongodb-integration-tests.md

```diff
@@ -0,0 +1,106 @@
+import { MongoDBServer } from 'mongomem';
+
+test.before('start mongodb server', async t => {
+  await MongoDBServer.start();
+})
+
+test('some feature', async t => {
+	const connectionString = await MongoDBServer.getConnectionString();
+	
+	// connectionString === 'mongodb://localhost:27017/3411fd12-b5d6-4860-854c-5bbdb011cb93'
+	// use connectionString to connect to the database with a client of your choice. See below for usage with Mongoose.
+});
+
+```
+
+## Cleaning Up
+
+After you have run your tests, you should include a `test.after.always` method to clean up the MongoDB server.
+This will remove any temporary files the server used while running.
+
+This is normally cleaned up by your operating system but it is good practise to do it manually to avoid OS-specific issues.
+
+```javascript
+test.after.always('cleanup', async t => {
+  MongoDBServer.tearDown(); // This will clean up temporary file storage.
+});
+```
+
+
+## Debugging
+If the server does not seem to start, you can set the following option before you call `MongoDBServer.start()`:
+`MongoDBServer.debug = true;`
+
+This will allow the MongoDB server to print connection or file permission errors when it's starting.
+It checks and picks an available port to run the server on, so errors are likely to be related to file permissions.
+
+## Extra: Setup and Use in Mongoose (MongoDB ODM)
+Mongoose is a robust Object-Document-Mapper (ODM) for MongoDB.
+
+Refer to [Mongoose ODM v4.9.4](http://mongoosejs.com/index.html) for full guides and documentation to get started with Mongoose.
+
+### Import Mongoose
+
+```javascript
+// myTestCase.test.js - (your test case file!)
+import mongoose from 'mongoose';
+```
+
+`mongoose` in this case is a single instance of the Mongoose ODM and is globally available. This is great for your application as it maintains a single access point to your database, but less great for isolated testing. 
+
+You should isolate Mongoose instances between your tests, so that the order of test execution is never depended on.
+This can be done with a little bit of work.
+
+### Isolate Mongoose Instance
+
+You can easily request a new instance of Mongoose.
+First, call `new mongoose.Mongoose()` to get the new instance, and then call `connect` with a database connection string provided by the `mongomem` package.
+
+**You will have to manually copy models from the global instance to your new instance.**
+
+```javascript
+import mongoose from 'mongoose';
+import { MongoDBServer } from 'mongomem'
+
+test.before('start mongodb', async t => {
+  await MongoDBServer.start();
+});
+
+test.beforeEach(async t => {
+  const db = new mongoose.Mongoose();
+    await db.connect(await MongoDBServer.getConnectionString());
+  
+    for (const name of mongoose.modelNames()) {
+      db.model(name, mongoose.model(name).schema);
+    }
+  
+    t.context.db = db;
+});
+
+test('my mongoose model integration test', async t => {
+	const {db} = t.context;
+      // do something…
```

Comment:
Too much indentation here as well.

The "do something" comment is a bit flippant (my bad!). Perhaps something like `// Now use the isolated DB instance in your test`.